services/core-api/app/api/projects/information_requirements_table/models/information_requirements_table.py

```diff
@@ -137,6 +158,14 @@ def update(self, irt_data, import_file=None, document_guid=None, add_to_session=
         if add_to_session:
             self.save()
 
+        if mine_doc and project:
+            if Config.ENVIRONMENT_NAME != 'prod':
+                renotify_hours = 24
+                mine = Mine.find_by_mine_guid(mine_doc.mine_guid)
+
+                trigger_notification(f'File(s) in project {project.project_title} has been updated for mine {mine_doc.mine_name}.',
```

Comment:
Are we calling trigger_notification twice? Once here and once in mine_document_resource.py? I'm guessing that the renotify param would prevent it from generating 2 notifications.

On the one hand, I like how the IRT is handling its own notifications (separation of concerns!) but on the other hand I don't like essentially the same method being copy-pasted into multiple classes (don't repeat yourself!) so I am on the fence about which is preferable.

Reply:
Well, `information_requirement_table`, `major_mine_application`, `project_dicesion_package` and `project_summary` all these call `trigger_notification` function as those are different file uploads. File archive is captured in mine_document_resource, only.

Reply:
Hmm, I'm wondering if there's a way we can reduce the repetition. Is the only difference between these functions the class name that's being queried? If so, I think that could be passed into a helper function (maybe in a project file?).

Reply:
You could maybe add a universal function for this in `mine_document_resource`, but I don't know how much benefit there would be there.  

While these would all be doing the same function, there are enough unique elements in each, that creating the generic function is going require either a bunch of props passed in or do a bunch of checks.  Personally I think I'd be more in favour of the minor repetition.  A few more lines of code, but it's also easier to read.

Reply:
I'd be satisfied with the really consistent parts being in a function.
- the renotify period
- the message text 
- optionally the feature flag check

It would be a _short_ function to have just have renotify_hours = 24 and the call to trigger_notification, but then if we want to change the time period, recipients, or the text we don't have to do it in 5 places.

Reply:
@taraepp I get your point. Actually `renotify_period` is already added as an optional param to `trigger_notification` function in `utils.py`. So only major repeating param is the text. Is there a way this text can be read from a file where we keep constants, so we can avoid the repetition?